File: PRACTICA 3/PYTHON/Pila.py
class Pila:
    def __init__(self, n):
        self.n = n
        self.arreglo = [0] * n
        self.top = -1

    def push(self, e):
        if self.isFull():
            print("Error: Pila llena")
            return
        self.top += 1
        self.arreglo[self.top] = e

    def pop(self):
        if self.isEmpty():
            print("Error: Pila vacía")
            return None
        valor = self.arreglo[self.top]
        self.top -= 1
        return valor

    def peek(self):
        if self.isEmpty():
            print("Error: Pila vacía")
            return None
        return self.arreglo[self.top]

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1


if __name__ == "__main__":
    pila = Pila(5)
    pila.push(10)
    pila.push(20)
    pila.push(30)
    print("Tope:", pila.peek())
    print("Elemento extraído:", pila.pop())
    print("Tope actual:", pila.peek())
    print("¿Está vacía?", pila.isEmpty())

File: PRACTICA 3/JAVA/Pila.java
public class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.top = -1;
    }

    public void push(long e) {
        if (isFull()) {
            System.out.println("Error: Pila llena");
            return;
        }
        arreglo[++top] = e;
    }

    public long pop() {
        if (isEmpty()) {
            System.out.println("Error: Pila vacía");
            return -1;
        }
        return arreglo[top--];
    }

    public long peek() {
        if (isEmpty()) {
            System.out.println("Error: Pila vacía");
            return -1;
        }
        return arreglo[top];
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }
}
